react/features/invite/components/InviteButton.web.js

```diff
@@ -0,0 +1,220 @@
+     * @param {Object} nextProps - The read-only props which this Component will
+     * receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (this.props._isDialOutAvailable !== nextProps._isDialOutAvailable
+            || this.props._isAddToCallAvailable
+                !== nextProps._isAddToCallAvailable) {
+            this._updateInviteItems(nextProps);
+        }
+    }
+
+    /**
+     * Renders the content of this component.
+     *
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t } = this.props;
+
+        const { VERTICAL_FILMSTRIP } = interfaceConfig;
+
+        return (
+            <div className = 'filmstrip__invite'>
+                <div className = 'invite-button-group'>
+                    <Button
+                        onClick = { this._onInviteClick }
+                        shouldFitContainer = { true }>
+                        { t('invite.invitePeople') }
+                    </Button>
+                    <DropdownMenu
+                        items = { this.state.inviteOptions }
+                        onItemActivated = { this._onInviteOptionSelected }
+                        position = { VERTICAL_FILMSTRIP
+                            ? 'bottom right'
+                            : 'top right' }
+                        shouldFlip = { true }
+                        triggerType = 'button' />
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Handles the click of the invite button.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onInviteClick() {
+        this.props.openDialog(InviteDialog);
+    }
+
+    /**
+     * Handles selection of the invite options.
+     *
+     * @param { Object } option - The invite option that has been selected from
+     * the dropdown menu.
+     * @private
+     * @returns {void}
+     */
+    _onInviteOptionSelected(option) {
+        this.state.inviteOptions[0].items.map(item => {
+            if (item.content === option.item.content) {
+                item.action();
+
+                return true;
+            }
+
+            return false;
+        });
+    }
+
+    /**
+     * Updates the invite items list depending on the availability of the
+     * features.
+     *
+     * @param {Object} props - The read-only properties of the component.
+     * @private
+     * @returns {void}
+     */
+    _updateInviteItems(props) {
+        const { t } = this.props;
+
+        const inviteItems = [
+            {
+                content: t('invite.shareTheLink'),
+                action: () => this.props.openDialog(InviteDialog)
+            }
+        ];
+
+        if (props._isDialOutAvailable) {
+            inviteItems.splice(0, 0, {
+                content: t('dialOut.dialOut'),
+                action: () => this.props.openDialog(DialOutDialog)
+            });
+        }
+
+        if (props._isAddToCallAvailable) {
+            inviteItems.splice(0, 0, {
```

Comment:
Why did you choose to do splice instead of unshift or maybe initializing inviteItems as empty first?

Reply:
I don't remember anymore :)

Reply:
Now I changed the code I little bit and I use splice in order to position the items in the list in the way they have been defined in interfaceConfig.